### ISSUE TITLE
How to provide validation data for multitask classifier?

### ISSUE THREAD

**Issue by davidfstein** [keras-team/keras#14622]
I have a multitask classifier with two outputs. The model is defined as follows:
```
def create_model(shape, 
                activation1='relu',
                layer1=256,
                layer2=512,
                dropout1=0.3,
                weight1=1.,
                weight2=1.,
                lr = .001):
    # define model
    inputs = Input(shape=shape, name='main_input')
    main_branch = Dense(layer1, activation=activation1)(inputs)
    main_branch = Dropout(dropout1)(main_branch)
    main_branch = Dense(layer2, activation=activation1)(main_branch)
    nrar_branch = Dense(1, activation='sigmoid', name='nrar_output')(main_branch)
    nrahr_branch = Dense(1, activation='sigmoid', name='nrahr_output')(main_branch)
    
    model = tf.keras.Model(inputs=inputs, outputs=[nrar_branch, nrahr_branch])
    # compile model
    model.compile(loss={'nrar_output': masked_loss_function,
                        'nrahr_output': masked_loss_function},
                  loss_weights={'nrar_output': weight1,
                                'nrahr_output': weight2},
                  optimizer=tf.optimizers.Adam(learning_rate=lr),
                  metrics=[tf.keras.metrics.AUC(curve="PR")]
                 )
    return model
```

and loss:
```
def masked_loss_function(y_true, y_pred):
        mask_value = -1
        mask = K.cast(K.not_equal(y_true, mask_value), K.floatx())
        return K.binary_crossentropy(y_true * mask, y_pred * mask)
```

I want to provide validation data in the model.fit method, but I'm not sure how to do this. Right now I am trying 
```
model.fit({'main_input': X_t},
              {'nrar_output': y_t['NR.AR'], 'nrahr_output': y_t['NR.AhR']},
              epochs=100, batch_size=256,
              verbose=1,
              callbacks=[PlotProgress(entity='auc', num=129)],
              validation_data=(X_val, (y_val['NR.AR'], y_val['NR.AhR']) )
             )
```
However, I'm not sure if this is working as I expect it to. The learning curves look reasonable but I'm still not sure. Is this functionality supported and if so is this the correct way to achieve it?

**Comment by sushreebarsa** [comment]
@davidfstein Could you please refer to the [model training](https://keras.io/api/models/model_training_apis/) ,[built in methods](https://keras.io/api/models/model_training_apis/), try with the latest TF v2.7.0 ?Please let us know if it helps? Thanks!
